projects/proximity_matcher_webservice/module.nix

```diff
@@ -0,0 +1,96 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.proximity_matcher_webservice;
+in
+{
+  options.services.proximity_matcher_webservice = {
+    enable = lib.mkEnableOption "proximity_matcher_webservice";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.proximity-matcher;
+      description = "The package to use.";
+    };
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "127.0.0.1";
+      description = "The address the webservice should listen on.";
+    };
+    listenPort = lib.mkOption {
+      type = lib.types.port;
+      default = 5000;
+      description = "The port the webservice should listen on.";
+    };
+    hashesPicklePath = lib.mkOption {
+      type = lib.types.path;
+      description = "Path to the pickle with TLSH hashes.";
+    };
+    hashesPath = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Path to a file with TLSH hashes, formatted as one hash per line.
+
+        It is only used with the optimized version of the server.
+      '';
+    };
+    optimized = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = "Whether to use the optimized version of the server.";
+    };
+    workers = lib.mkOption {
+      type = lib.types.int;
+      default = 4;
+      description = "The number of workers gunicorn should use.";
+    };
+    timeout = lib.mkOption {
+      type = lib.types.int;
+      default = 60;
+      description = "The timeout in seconds for workers of the optimized server.";
+    };
+  };
+  config = {
+    systemd.services.proximity_matcher_webservice = lib.mkIf cfg.enable {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment = {
+        PROXIMITY_CONFIGURATION = pkgs.writeText "proximity_config.py" (
+          ''
+            TLSH_PICKLE_FILE = '${cfg.hashesPicklePath}'
+          ''
+          + lib.optionalString (!(isNull cfg.hashesPath)) ''
+            KNOWN_TLSH_HASHES = '${cfg.hashesPath}'
+          ''
+        );
+      };
+      serviceConfig = {
+        ExecStart = "${
+          lib.getExe' (pkgs.python3.withPackages (_: [ cfg.package ])) "gunicorn"
+        } -w ${toString cfg.workers} -b ${cfg.listenAddress}:${toString cfg.listenPort} --preload${lib.optionalString cfg.optimized " -t ${toString cfg.timeout}"} proximity_matcher_webservice.proximity_server${lib.optionalString cfg.optimized "_opt"}:app";
```

Comment:
```suggestion
        } -w ${toString cfg.workers} -b ${cfg.listenAddress}:${toString cfg.listenPort} --preload${lib.optionalString cfg.optimized " -t ${toString cfg.timeout}"} proximity-matcher.proximity_server${lib.optionalString cfg.optimized "_opt"}:app";
```

Reply:
This one refers to the name of the actual python module that was built, which uses the same name as the GitHub repo. So replacing it breaks things:

```
gunicorn[854]: ModuleNotFoundError: No module named 'proximity-matcher'
```